### ISSUE TITLE
Disallow instantiation if class has abstact attributes (even if there is no ABCMeta metaclass)

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#3127]
Pyright does not complain about the following code whereas mypy reports:

> Cannot instantiate abstract class "Foo" with abstract attribute "f"

```python
from abc import abstractmethod

class Foo:
    @abstractmethod
    def f(self) -> None:
        pass

Foo() # error
```

Typeshed uses this when the actual implementation doesn't use `ABCMeta` for classes that should be abstract (in which case adding `ABCMeta` in the type stub would be inaccurate since `ABCMeta.register` isn't available).

**Comment by erictraut** [comment]
This check is expensive to perform because it requires evaluating all of the methods in a class to determine whether they've been decorated with an `abstractmethod` decorator. I don't think this provides enough value to justify the analysis slowdown.
